tutorials/python-api.md

```diff
@@ -0,0 +1,108 @@
+# LitGPT High-level Python API
+
+This is a work-in-progress draft for a high-level LitGPT Pyhon API.
+
+&nbsp;
+## Model loading & saving
+
+The `LLM.load` command loads an `llm` object, which contains both the model object (a PyTorch module) and a preprocessor.
+
+```python
+from litgpt import LLM
+
+llm = LLM.load(
+    source="url | local_path",
+    # high-level user only needs to care about those:
+    memory_reduction="none | medium | strong"
+    # advanced options for technical users:
+    hub="hf | local | other"
+    quantize="bnb.nf4",
+    precision="bf16-true",
+    device="cuda | cpu",
+)
+```
+
+Here,
+
+-  `llm.model` contains the PyTorch Module
+- and `llm.preprocessor.tokenizer`  contains the tokenizer
+
+The `llm.save` command saves the model weights, tokenizer, and configuration information.
+
+
+```python
+llm.save(checkpoint_dir, format="lightning | ollama | hf")
+```
+
+
+&nbsp;
+## Inference / Chat
+
+```
+response = llm.generate(
+    prompt="What do Llamas eat?",
+    temperature=0.1,
+    top_p=0.8,
+    ...
+)
+```
+
+
+&nbsp;
+## Dataset
+
+The `llm.prepare_dataset` command prepares a dataset for training.
+
+```
+llm.download_dataset(
+    URL,
+    ...
+)
+```
+
+```
+dataset = llm.prepare_dataset(
```

Comment:
Would this call into this?
https://github.com/Lightning-AI/litgpt/pull/1057/files#diff-4306b5fd7fdf76fee60f7f51d66e02090a620df0c4e8883447cd9a2a559ed1d2
(I should probably finish it)
It is currently only needed for pretraining.

Reply:
Ideally yes :). But for the v1 that we need, we don't have to worry about pretraining and finetuning yet.